core/src/appleMain/kotlin/ScannerBuilder.kt

```diff
@@ -16,6 +16,14 @@ public actual class ScannerBuilder {
         get() = filters?.filterIsInstance<Filter.Service>()?.map { it.uuid }
 
     public actual var filters: List<Filter>? = null
+
+    /**
+     * Scanning options for Apple's CBCentralManager.scanForPeripherals(withServices:options:).
+     * This enables peripheral scanning options such as CBCentralManagerScanOptionAllowDuplicatesKey
+     * to be specified when scanning begins.
+     */
+    public var scanOptions: Map<Any?, *>? = null
```

Comment:
To make the options more discoverable (and less error-prone) it might make sense to have this be a `data class` that defines the available options. Then internally, it would be converted to the `Map<String, *>` that Core Bluetooth is expecting?

For example:

```kotlin
data class ScanOptions(
    val allowDuplicateKeys: Boolean,
    val solicitedServiceUuids: Array<Uuid>,
)
```

Then, in the builder we would have something along the lines of:

```suggestion
    /**
     * Scanning options for Apple's CBCentralManager.scanForPeripherals(withServices:options:).
     * This enables peripheral scanning options such as CBCentralManagerScanOptionAllowDuplicatesKey
     * to be specified when scanning begins.
     */
    public var options: ScanOptions? = null
```

Alternatively, since there are so few options, might make sense to have them be properties directly on the Scanner builder; I'm not quite sure which I like better.

What do you think?